### PR TITLE
Disable plugin SaveAs for non-ADMIN role users in Contexts

### PR DESCRIPTION
**Pull request by Igi-ID**
## Description
This PR enables visualizing the SaveAs plugin as a child of Save plugin in the list of available plugins in the context creation wizard. The SaveAs plugin is enabled by default. The visualization of the plugin allows for the context creator to configure it. Given the requirements of the issue noted below, the configuration is simply `cfg.disablePluginIf": "{state('userrole') !== 'ADMIN'}`.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Minor changes to existing features

[SaveAs plugin configuration docs](https://mapstore.geosolutionsgroup.com/mapstore/docs/api/plugins#plugins.SaveAs)

## Issue

**What is the current behavior?**
[#9919 ](https://github.com/geosolutions-it/MapStore2/issues/9919)

**What is the new behavior?**
SaveAs plugin is now a child of Save plugin in the list of available plugins in the context creation wizard. The SaveAs plugin is enabled by default. The visualization of the plugin allows for the context creator to configure it. Given the requirements of the issue noted below, the configuration is simply `cfg.disablePluginIf": "{state('userrole') !== 'ADMIN'}`.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/SaveAs.jsx [hunk 1]
```diff
@@ -24,6 +24,8 @@ const showMapSaveAsSelector = state => state.controls && state.controls.mapSaveA
 /**
  * Plugin for Create/Clone a Map. Saves the map as a new Resource (using the persistence API).
  * @prop {boolean} [cfg.disablePermission=false] disable the permission selector in the tool. Can be used in context when permissions are not needed (resources are private only/using plugin with another API)
+ * @prop {object} [cfg.disablePluginIf=Boolean(expression)] disable the plugin with a custom expression that evaluates to a boolean. Can be configured in context creation or when editing a context.
+ * You may use an expression that evaluates a particular state of the program. For example: `"disablePluginIf": "{state('userrole') !== 'ADMIN'}"`. In the example, the plugin is disabled for users that do not have the role of ADMIN.
```
Comment (offtherailz) [inline web/client/plugins/SaveAs.jsx:28]: This property is valid for all the plugins and already documented. We don't need to add this doc here.

```suggestion
```

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV QA ? Thank you. See in the issue how to test, configuring in the SaveAs plugin the cfg optin `"disablePluginIf": "{state('userrole') !== 'ADMIN'}"`, the saveAs plugin should be visible only for administrators.

**Comment by ElenaGallo** [comment]
Hi @Igi-ID @offtherailz,
on map context the pr works as expected but if I insert the cfg option "disablePluginIf": "{state('userrole') !== 'ADMIN'}" in the SaveAs plugin, when I open the context with a user with edit permission also the save button is not visible on the sidebar.

*Without the cfg option in the SaveAs plugin *

Open [this context](https://github.com/geosolutions-it/MapStore2/assets/56537133/0a7bac3b-ccd9-46e5-90eb-ffd34cebbeb8) with a user in the geosolutions group

https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/saveas
*With the cfg option in the SaveAs plugin*

https://github.com/geosolutions-it/MapStore2/assets/56537133/3301d6f6-6856-425e-8a67-82a211f14de4

Open [this context](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/pluginSaveAs) with a user in the geosolutions group

**Comment by offtherailz** [comment]
Save is not visible because you are in a context, while the save button saves a map.
Contexts are map viewers that can be configured with various plugins, and can include or not include the context of map to save etc...
Their URLs are something like: 
https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/saveas

Contexts can also be used to display a map, in this case the URL will be 

https://dev-mapstore.geosolutionsgroup.com/mapstore/#/context/saveas/48665

This map creation for a context can be done:
- From a context, using "save as" plugin
- From the home page
- From a map in a context, usin "save as" plugin (or save, if you want to update the current map).

So a possible sequence could be this:
create a context 
- "Save as" plugin should be visible if configured user can 
- "Save" should never be visible, because you are not in a map.

When you create a map: 
- "Save as" plugin should be visible if configured user can 
- "Save" should be visible accordingly to configuration and map edit permission

This is the actual behavior. 
If you create a map using save as, you will see both. Could you please confim

**Comment by ElenaGallo** [comment]
@offtherailz yes, I confirm. Thank you for the clarification. I'm closing the issue.

**Pull request merged**